nixos/tests/common/acme/server/default.nix

```
@@ -100,28 +60,46 @@ in
   };
 
   config = {
-    test-support = {
-      resolver.enable =
-        let
-          isLocalResolver = config.networking.nameservers == [ "127.0.0.1" ];
-        in
-        lib.mkOverride 900 isLocalResolver;
-    };
+    networking = {
+      firewall.allowedTCPPorts = [
+        80
+        443
+        15000
+        4002
+      ];
 
-    # This has priority 140, because modules/testing/test-instrumentation.nix
-    # already overrides this with priority 150.
-    networking.nameservers = lib.mkOverride 140 [ "127.0.0.1" ];
-    networking.firewall.allowedTCPPorts = [
-      80
-      443
-      15000
-      4002
-    ];
+      # Match the caDomain - nixos/lib/testing/network.nix will then add a record for us to
+      # all nodes in /etc/hosts
+      hostName = "acme";
+      domain = "test";
 
-    networking.extraHosts = ''
-      127.0.0.1 ${domain}
-      ${config.networking.primaryIPAddress} ${domain}
-    '';
+      # Extend /etc/hosts to resolve all configured certificates to their hosts.
+      # This way, no DNS server will be needed to validate HTTP-01 certs.
+      hosts = lib.attrsets.concatMapAttrs (_: node: let
+        inherit (node.networking) primaryIPAddress primaryIPv6Address;
+        ips = builtins.filter (ip: ip != "") [primaryIPAddress primaryIPv6Address];
```

To match the rest of the file:
```suggestion
        ips = builtins.filter (ip: ip != "") [ primaryIPAddress primaryIPv6Address ];
```